org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/XMLSymbolsProvider.java

```diff
@@ -169,12 +175,16 @@ private void findSymbolInformations(DOMNode node, String container, List<SymbolI
 			Range range = getSymbolRange(node);
 			Location location = new Location(xmlDocument.getDocumentURI(), range);
 			SymbolInformation symbol = new SymbolInformation(name, getSymbolKind(node), location, container);
+			
 			symbols.add(symbol);
+			if (symbols.size() > limit) {

```

Comment:
The computation will stop if `symbols.size() > limit` instead of `symbols.size() >= limit`. This is because I wanted a way to differentiate two situations: when the number of symbols matches the limit, and when the number of symbols exceeds the limit.

I currently have it so that there will be no notification sent if the number matches the limit. It is sent only when it exceeds the limit.

Reply:
@xorye see my draft PR which throws an exception to stop the process of compute of symbols

Reply:
@angelozerr thank you!